src/soc/opentitan/earlgrey/src/bootblock.S

```diff
@@ -21,6 +21,92 @@
 _boot:
 	// These codes need to be implemented on a specific SoC.
 //	.weak cache_as_ram
+	# The previous boot stage passes these variables:
+	#   a0: hartid
+	#   a1: ROM FDT
+	# a0 is redundant with the mhartid register. a1 might not be valid on
+	# some hardware configurations, but is always set in QEMU.
+	csrr a0, mhartid
+
+	# Reset the mtimecmp to some absurdly large number.
+	# The mtimecmp register is memory mapped at 0x2004000 + 0x8 * mhartid
+	# TODO: Move to Rust
+	li t0, 0x2004000
+	slli t1, a0, 3
+	add t1, t1, t0
+	li t0, 0xfffffff
+	# sd t0, (t1)
+
+	# Clear all interrupt enable and pending bits.
+	# TODO: Move to Rust
+	csrwi mie, 0x0
+	csrwi mip, 0x0
+
+	# TODO: fix this comment and figure out RAM sizes.
+	# Layout of L2-LIM (address 0x08000000):
+	#   0x00000-0x0ffff: .bss segment (64K bytes)
+	#   0x10000-0x1ffff: stack of boot hart (64K bytes)
+	#   0x20000-0x20fff: stack of hart 0 (4K bytes)
+	#   0x21000-0x21fff: stack of hart 1 (4K bytes)
+	#   ...
+	#   0x2n000-0x2nfff: stack of hart n (4K bytes)
+
+	# Clear the .bss segment. Every hart does this, which should be safe.
+	li t0, 0x08000000
+	li t1, 0x08010000
+	li t1, 0x08000100
+clear_bss:
+	# verilator crashes and burns on the first sw.
+	sw zero, (t0)
+	addi t0, t0, 8
+	bne t0, t1, clear_bss
+
+	# Decide if this is the boot hart.
```

Comment:
There's only one hart in the OpenTitan design, which can simplify the startup code here.